src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -6,6 +9,39 @@ namespace Datadog.Trace.Configuration
     /// </summary>
     public abstract class StringConfigurationSource : IConfigurationSource
     {
+        /// <summary>
+        /// Returns a <see cref="IDictionary{TKey, TValue}"/> from parsing
+        /// <paramref name="data"/>.
+        /// </summary>
+        /// <param name="data">A string containing key-value pairs which are comma-separated, and for which the key and value are colon-separated.</param>
+        /// <returns><see cref="IDictionary{TKey, TValue}"/> of key value pairs.</returns>
+        public static IDictionary<string, string> ParseCustomKeyValues(string data)
+        {
+            var dictionary = new ConcurrentDictionary<string, string>();
+
+            if (data == null)
```

Comment:
Perhaps we can use the `string.IsNullOrWhitespace` API to exit early

Reply:
for sure. thanks